NATS.Client.Core/Internal/InboxSub.cs

```diff
@@ -38,7 +55,11 @@ protected override void TryComplete()
 internal class InboxSubBuilder : ISubscriptionManager
 {
     private readonly ILogger<InboxSubBuilder> _logger;
+#if NET6_0_OR_GREATER
     private readonly ConcurrentDictionary<string, ConditionalWeakTable<NatsSubBase, object>> _bySubject = new();
```

Comment:
You can keep using the `ConditionalWeakTable` and instead of `AddOrUpdate` you can on NS2.0 call `subTable.CreateObject(s, _ => a_singleton)`. You can pass a singleton to `AddOrUpdate` as well.